sdk/storage/azure-storage-blob/tests/test_common_blob.py

```diff
@@ -2788,21 +2788,20 @@ def test_download_to_file_with_existing_file(self, **kwargs):
         self._setup(storage_account_name, storage_account_key)
         data = b'123' * 1024
         source_blob = self._create_blob(data=data)
-        file_path = 'file_with_existing_file.temp.{}.dat'.format(str(uuid.uuid4()))
 
         # Act
-        download_blob_from_url(
-            source_blob.url, file_path,
-            credential=storage_account_key)
+        with tempfile.NamedTemporaryFile(delete=False) as temp_file:
+            download_blob_from_url(source_blob.url, temp_file.name, credential=storage_account_key, overwrite=True)
 
-        with pytest.raises(ValueError):
-            download_blob_from_url(source_blob.url, file_path)
+            with pytest.raises(ValueError):
+                download_blob_from_url(source_blob.url, temp_file.name)
 
-        # Assert
-        with open(file_path, 'rb') as stream:
-            actual = stream.read()
+            # Assert
+            temp_file.seek(0)
+            actual = temp_file.read()
             assert data == actual
-        self._teardown(file_path)
+
+            temp_file.close()
```

Comment:
I think to delete the file we need to do something additional? I'm seeing `os.unlink(temp_file)`? Maybe also throw that in a try/finally block so well always unlink the file even in the case of an error?